Sources/SwiftParser/Types.swift

```diff
@@ -689,7 +684,7 @@ extension Parser.Lookahead {
       // Handle type-function if we have an '->' with optional
       // 'async' and/or 'throws'.
       var loopProgress = LoopProgressCondition()
-      while let (_, handle) = self.at(anyIn: EffectsSpecifier.self), loopProgress.evaluate(currentToken) {
+      while let (_, handle) = self.at(anyIn: EffectSpecifier.self), loopProgress.evaluate(currentToken) {
```

Comment:
We also handle unexpected up to the `throws` from `canRecoverTo`. We don't beyond that point though (see other comment).